Enigmatry.Entry.AspNetCore.Authorization/Requirements/UserHasPermission.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Enigmatry.Entry.AspNetCore.Authorization.Attributes;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Enigmatry.Entry.AspNetCore.Authorization.Requirements;
+
+#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
+public static class UserHasPermission
+#pragma warning restore CA1711 // Identifiers should not have incorrect suffix
+{
+    public class Requirement : IAuthorizationRequirement { }
```

Comment:
The UserHasPermission.Requirement feels a bit strange without any properties. I would expect it to contain a list of Permissions that must be validated in the handler. Instead we need to search for the UserHasPermissionAttribute to get the list.

I checked the documentation, there is something similar to what we are trying to do:
https://learn.microsoft.com/en-us/aspnet/core/security/authorization/iauthorizationpolicyprovider?view=aspnetcore-7.0#custom-authorization-attributes

In the example they map the AuthorizationAttribute parameters to a policy name. Afterwards parse policy name to get parameters to construct the policy requirement. It looks weird, but probably worth investigating.

Btw, concatenated string Permissions in the UserHasPermissionAttribute feels like a policy name.